runtime/libffi/z/ffi64.c

```diff
@@ -766,6 +768,50 @@ ffi_prep_args (unsigned char *stack, extended_cif *ecif)
 
 /*======================== End of Routine ============================*/
 
+/**
+ * Helper functions to know if the given struct needs to be treated for complex
+ * type for float or double
+ */
+unsigned short
+get_ffi_element_type_in_struct(ffi_type *arg_type)
+{
+  while ((FFI_TYPE_STRUCT == arg_type->type)
+        && (NULL != arg_type->elements[0])
+        && (NULL == arg_type->elements[1])
+  ) {
+    arg_type = arg_type->elements[0];
+  }
+  return arg_type->type;
+}
+
+unsigned short
+ffi_check_struct_for_complex(ffi_type *arg_type)
+{
+  if (FFI_TYPE_STRUCT == arg_type->type)
```

Comment:
Should this also check that the struct is not empty?
```c
  if ((FFI_TYPE_STRUCT == arg_type->type) && (NULL != arg_type->elements[0]))
```

Reply:
Yes, It should check that, I missed that - Fixed in https://github.com/eclipse-openj9/openj9/compare/6322f343e9e88af4b64f7d08586a47a7ecb89b32..583867825be4477a8319f21ffb952780631fbeb3